modules/python3qt/src/pythonmenu.cpp

```diff
@@ -60,106 +60,212 @@
 #include <QUrl>              // for QUrl, QUrl::TolerantMode
 #include <Qt>                // for WA_DeleteOnClose
 #include <QMenuBar>          // for QMenuBar
-#include <fmt/core.h>        // for basic_string_view, arg
-#include <fmt/ostream.h>     // for print
+#include <QInputDialog>
+#include <fmt/core.h>     // for basic_string_view, arg
+#include <fmt/ostream.h>  // for print
 #include <fmt/std.h>
 
 class QObject;
 
 namespace inviwo {
 
-PythonMenu::PythonMenu(InviwoModule* pymodule, InviwoApplication* app) {
-    if (auto win = utilqt::getApplicationMainWindow()) {
-
-        auto newEditor = [this, win, app]() {
-            auto editor = std::make_unique<PythonEditorWidget>(win, app);
-            editor->loadState();
-            editor->setAttribute(Qt::WA_DeleteOnClose);
-            if (!editors_.empty()) {
-                auto newPos = editors_.back()->pos() + QPoint(40, 40);
-                if (newPos.x() > 800) newPos.setX(350);
-                if (newPos.y() > 800) newPos.setX(100);
-                editor->move(newPos);
+PythonMenu::PythonMenu(const std::filesystem::path& modulePath, InviwoApplication* app,
+                       QMainWindow* win)
+    : app_{app}
+    , win_{win}
+    , editors_{}
+    , toolbar_{win_->addToolBar("Python")}
```

Comment:
guaranteed for `win` not being a nullptr?

Reply:
We have assumed that in the rest of the code at least...